packages/utils/sets-utils/src/cell-set-utils.js

```diff
@@ -559,6 +559,76 @@ export function filterNode(node, prevPath, filterPath) {
   };
 }
 
+/**
+ * Two arrays of paths are equal (contain the same elements).
+ * @param {Array<string[]>} a Set of paths.
+ * @param {Array<string[]>} b Other set of paths.
+ * @returns {boolean} Whether the two sets are equivalent.
+ */
+export function isEqualSet(a, b) {
+  if (a.length === b.length) {
+    const aSet = new Set(a);
```

Comment:
Why is this needed? For `delete`? It doesn't seem like this would ever actually dedup something (at least the behavior is not tested)

Reply:
These functions are needed to check whether the parent checkbox should be:
- checked (`isFullSubSet` true - all of the parent's children are selected)
- indeterminate (`isPartialSet && !isEqualSet` true - subset of the parent's children are selected)
- unchecked (otherwise - none of the parent's children)

They are implemented using `isEqual` because the set elements are arrays of strings like `["Clustering", "Cluster 1"]` so normal equality operators won't work.

<img width="361" alt="Screen Shot 2023-01-05 at 10 14 51 AM" src="https://user-images.githubusercontent.com/7525285/214948261-0ae49f15-8feb-4a16-8426-959baaef8f4d.png">